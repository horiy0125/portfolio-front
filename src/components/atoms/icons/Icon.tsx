import React from 'react';
import styled from 'styled-components';
import { colors } from '../../../styles/variables';
import { classes } from '../../../utils/classes';

type Props = {
  icon: React.ReactNode;
  className?: string;
};

const Root = styled.span`
  display: inline-block;
  width: 24px;
  height: 24px;
`;

const Icon: React.VFC<Props> = ({ icon, className }) => {
  return <Root className={classes(className)}>{icon}</Root>;
};

export default Icon;
