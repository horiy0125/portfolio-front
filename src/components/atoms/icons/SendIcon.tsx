import React from 'react';

const SendIcon: React.VFC = () => {
  return (
    <svg
      className="MuiSvgIcon-root jss81"
      focusable="false"
      viewBox="0 0 24 24"
      aria-hidden="true"
    >
      <path d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"></path>
    </svg>
  );
};

export default SendIcon;
