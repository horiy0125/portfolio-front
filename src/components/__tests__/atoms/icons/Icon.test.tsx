import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../../../atoms/icons/Icon';
import SendIcon from '../../../atoms/icons/SendIcon';

describe('components/atoms/icons/Icon', () => {
  describe('if the components has svg icon', () => {
    it('render svg icon as children', () => {
      const component = renderer.create(<Icon icon={<SendIcon />} />);
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('if the props `icon` is null', () => {
    it('render only myself', () => {
      const component = renderer.create(<Icon icon={null} />);
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
