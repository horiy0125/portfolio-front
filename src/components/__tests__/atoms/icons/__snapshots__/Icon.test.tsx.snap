// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/icons/Icon if the components has svg icon render svg icon as children 1`] = `
<span
  className="sc-bdnxRM icjhvk"
>
  <svg
    aria-hidden="true"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
    />
  </svg>
</span>
`;

exports[`components/atoms/icons/Icon if the props \`icon\` is null render only myself 1`] = `
<span
  className="sc-bdnxRM icjhvk"
/>
`;
